### PR TITLE
Add autoclave cipher

### PR DESCRIPTION
**Pull request by VictorRS27**
### Describe your change:

I'm adding the autoclave or autokey cipher.
Is similar to the vigenerè cipher, but using the own plaintext as the key.
The code implements an encrypt and a decrypt function.
For more info:
https://en.wikipedia.org/wiki/Autokey_cipher

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: ciphers/autoclave.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    Function that encrypt a given plaintext (string)
+    and key (string), returning the encrypted ciphertext
+
+    @params
+    plaintext - a normal text to be encrypted (string)
+    key - a small text or word to start the replacing (sFtring)
+
+    @return
+    A string with the ciphertext
+
```
Comment (CaedenPH) [inline ciphers/autoclave.py:12]:
```suggestion
```
Typically, we want the parameter names to be self-documenting so this isn't really necessary

File: ciphers/autoclave.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    'jsqqs avvwo'
+    """
+    if type(plaintext) != str:
+        raise TypeError("plaintext must be a string")
+    if type(key) != str:
+        raise TypeError("key must be a string")
+
+    if plaintext == "":
+        raise ValueError("plaintext is empty")
+    if key == "":
```
Comment (CaedenPH) [inline ciphers/autoclave.py:23]:
```suggestion
    if not plaintext:
        raise ValueError("plaintext is empty")
    if not key:
```

File: ciphers/autoclave.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        raise ValueError("key is empty")
+
+    key = key + plaintext
```
Comment (CaedenPH) [inline ciphers/autoclave.py:26]:
```suggestion
    key += plaintext
```

File: ciphers/autoclave.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
+def encrypt(plaintext, key) -> str:
```
Comment (CaedenPH) [inline ciphers/autoclave.py:1]:
```suggestion
def encrypt(plaintext, key) -> str:
```
These parameters need to be typehinted

File: ciphers/autoclave.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+operation = int(input("Type 1 to encrypt or 2 to decrypt:"))
+if operation == 1:
+    plaintext = str(input("Typeplaintext to be encrypted:\n"))
+    key = str(input("Type the key:\n"))
+    print(encrypt(plaintext, key))
+elif operation == 2:
+    ciphertext = str(input("Type the ciphertext to be decrypted:\n"))
+    key = str(input("Type the key:\n"))
+    print(decrypt(ciphertext, key))
+decrypt("jsqqs avvwo", "coffee")
```
Comment (CaedenPH) [inline ciphers/autoclave.py:111]: please move this into an `if __name__ == "__main__":` block

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please link some documentation at the start of the file and include more doctests within each function

File: ciphers/autoclave.py (hunk 4 above)
Comment (cclauss) [inline ciphers/autoclave.py:1]: Clave is Latin for key so the more popular name for this algorithm is autokey.
```suggestion
”””
https://en.wikipedia.org/wiki/Autokey_cipher
> An autokey cipher (also known as the autoclave cipher) is a cipher that incorporates the message (the plaintext into the key. The key is generated from the message in some automated fashion, sometimes by selecting certain letters from the text or, more commonly, by adding a short primer key to the front of the message.
”””


def encrypt(plaintext, key) -> str:
```

File: ciphers/autoclave.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    plaintext - a normal text to be encrypted (string)
+    key - a small text or word to start the replacing (sFtring)
+
+    @return
+    A string with the ciphertext
+
+    >>> encrypt("hello world", "coffee")
+    'jsqqs avvwo'
+    """
+    if type(plaintext) != str:
```
Comment (cclauss) [inline ciphers/autoclave.py:16]: PEP8: Use `isinstance(plaintext, str)` instead of directly comparing types.

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
@priya-sundaram-dev Several years ago, when I merged this pull request, I disabled five unrelated algorithms.  I have no idea why I disabled them, but perhaps they were slowing down our CI tests.  Can you please create a pull request that re-enables them and also re-enables neural_network/perceptron.py.DISABLED, as discussed in:
* #15206

**Comment by priya-sundaram-dev** [comment]
Done — opened #15208. It re-enables the four ML examples plus `neural_network/perceptron.py`.

The reason they wouldn't come back as-is: they'd bit-rotted against scikit-learn. `plot_confusion_matrix` and the Boston dataset were both removed in scikit-learn 1.2, so `--doctest-modules` couldn't even import the files. I swapped in `ConfusionMatrixDisplay.from_estimator` and `load_diabetes`, dropped the artificial `time.sleep()`s, and for the perceptron I seeded a local RNG + capped the training loop so it can't spin forever on non-separable data. All doctests + ruff pass locally.

**Pull request merged**